### PR TITLE
Switch CI/CD to S3

### PR DESCRIPTION
**Pull request by FedericoCeratto**

### CODE REVIEW

File: .circleci/debops-ci [hunk 1]
```diff
@@ -1,40 +1,77 @@
...
 build             - locate and build packages
-upload <filename> - upload one package to Bintray
-ci                - detect CircleCI PRs, build and upload packages to Bintray
-delete_from_bintray <filename> - delete filename from Bintray
+upload <filename> - upload one package to S3 or Bintray
+ci                - detect CircleCI PRs, build and upload packages
+delete_from_archive <filename> - delete filename from archive
+
+Features:
+ - Implent CI/CD workflow using package archives
```
Comment (hellais) [inline .circleci/debops-ci]:
```suggestion
 - Implement CI/CD workflow using package archives
```

File: .circleci/debops-ci [hunk 2]
```diff
@@ -73,42 +121,81 @@ def trim_compare(url: str) -> str:
...
+
+    else:
+        # This is a build for a new branch but without a PR: ignore it
+        return []
+
+
+def _set_pkg_version_from_github_actions(p, ver):
+    # FIXME
+    print(getenv("GITHUB_EVENT_PATH"))
+    comp = trim_compare(getenv("CIRCLE_COMPARE_URL", ""))  # show changes in VCS
```
Comment (hellais) [inline .circleci/debops-ci]: Shouldn't these env variables be specific to github?

Comment (hellais) [reply .circleci/debops-ci]: Ah maybe that is why there is the FIXME comment.

File: .circleci/debops-ci [hunk 3]
```diff
@@ -150,62 +286,252 @@ def build() -> None:
...
+        )
+        with open(tf.name, "w") as f:
+            f.write(html)
+
+        r, o = runc(f"{put} {tf.name} {baseuri}/index.html")
+
+    def lock(self, conf, baseuri):
+        """Rename semaphore file"""
+        print(f"Locking {baseuri} ...")
+        cmd = f"s3cmd mv --no-progress {baseuri}/.debrepos3.nolock {baseuri}/.debrepos3.lock"
```
Comment (hellais) [inline .circleci/debops-ci:431]: Be careful that s3 renaming is not atomic. s3cmd implement the `mv` operation as doing a copy and then a delete: https://github.com/s3tools/s3cmd/blob/5834228d5bddde3819a3ae5506dd656b62dfb2d1/S3/S3.py#L1007

Comment (FedericoCeratto) [reply .circleci/debops-ci:431]: Yes, the fact that there are 2 copies of the file for a short time should act as a 2-step lock mechanism.

**Review by hellais — APPROVED** [review]
LGTM

**Pull request merged**